File: Onboarding_client/src/app/models/checklist.ts
import {Item} from "./item";


export class Checklist {
  id: string;
  abteilungsname: string;
  position: string;
  items: Item[];

  constructor(
    id: string,
    abteilungsname: string,
    position: string,
    items: Item[]
  ) {
    this.id = id;
    this.abteilungsname = abteilungsname;
    this.position = position;
    this.items = items;
  }
}

File: src/main/java/Raiffeisen/Onboarding/Entities/CheckList.java
package Raiffeisen.Onboarding.Entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.GenericGenerator;

import java.util.List;

@Entity
@Table(name="c_checklist")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
public class CheckList {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID, generator = "uuid2")
    @GenericGenerator(name="uuid2", strategy = "uuid2")
    @Column(name="c_id")
    private String id;

    @Column(name = "c_Position")
    private String position;

    @Column(name = "c_abteilungsname")
    private String abteilungsname;

    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.DETACH)
    @JoinColumn(name = "c_i_id")
    private List<Item> items;
}